conductor-client/src/main/kotlin/com/openlattice/collaborations/PostgresCollaborationDatabaseService.kt

```diff
@@ -0,0 +1,256 @@
+
+        PostgresProjectionService.importTableFromFdw(
+                hds = collaborationHds,
+                fdwName = getFdwName(organizationId),
+                sourceSchema = foreignSchema,
+                sourceTableName = foreignName,
+                destinationSchema = intermediateSchema,
+                destinationTableName = intermediateName
+        )
+
+        PostgresProjectionService.createViewOverTable(
+                hds = collaborationHds,
+                sourceTableSchema = intermediateSchema,
+                sourceTableName = intermediateName,
+                viewSchema = viewSchema,
+                viewName = viewName
+        )
+
+        externalDbPermissioner.initializeProjectedTableViewPermissions(
+                collaborationId,
+                viewSchema,
+                table,
+                columns
+        )
+    }
+
+    override fun removeTableProjection(collaborationId: UUID, organizationId: UUID, tableId: UUID) {
+        logger.info("destroy the thing!")
+
+        val table = externalTables.getValue(tableId)
+        val orgSchema = externalDbConnMan.getOrganizationDatabaseName(organizationId)
+        val collaborationHds = externalDbConnMan.connectToOrg(collaborationId)
+
+        PostgresProjectionService.destroyViewOverTable(collaborationHds, orgSchema, table.name)
+        PostgresProjectionService.dropTableImportedFromFdw(collaborationHds, Schemas.PROJECTIONS_SCHEMA.label, table.id.toString())
+    }
+
+    override fun refreshTableProjection(collaborationId: UUID, organizationId: UUID, tableId: UUID) {
+        removeTableProjection(collaborationId, organizationId, tableId)
+        initializeTableProjection(collaborationId, organizationId, tableId)
+    }
+
+    private fun createOrganizationFdw(collaborationId: UUID, organizationId: UUID) {
+        val organizationDbName = externalDbConnMan.getOrganizationDatabaseName(organizationId)
+        val fdwName = getFdwName(organizationId)
+
+        val atlasUsername = assemblerConfiguration.server.getProperty("username")
+        val atlasPassword = assemblerConfiguration.server.getProperty("password")
+
+        val collaborationHds = externalDbConnMan.connectToOrg(collaborationId)
+        val orgDbJdbc = externalDbConnMan.appendDatabaseToJdbcPartial(
+                assemblerConfiguration.server.getProperty("jdbcUrl"),
+                organizationDbName
+        )
+
+        PostgresProjectionService.createFdwBetweenDatabases(
+                localDbDatasource = collaborationHds,
+                remoteUser = atlasUsername,
+                remotePassword = atlasPassword,
+                remoteDbJdbc = orgDbJdbc,
+                localUsername = atlasUsername,
+                localSchema = Schemas.PROJECTIONS_SCHEMA,
+                fdwName = fdwName
+        )
+    }
+
+    private fun getFdwName(organizationId: UUID): String {
+        return "fdw_${organizationId.toString().replace("-", "")}"
```

Comment:
Seems like this should be in PostgresProjectionService

Reply:
Then again, maybe not. I'm flip-flopping

Reply:
ehh I prefer keeping PostgresProjectionService totally agnostic, and managing collaboration-specific stuff in here